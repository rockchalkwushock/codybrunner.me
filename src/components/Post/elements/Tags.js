import React from 'react'
import styled from 'styled-components'
import { arrayOf, string } from 'prop-types'

import Link from '../../Link'
import { Tag } from '../../Card'

const Tags = styled.ul`
  display: grid;
  grid-gap: 5px;
  grid-template-columns: repeat(auto-fill, minmax(80px, 1fr));
`

Tags.displayName = 'Tags'

const PostTags = ({ tags }) => (
  <Tags>
    {tags.map(t => (
      <Tag key={t}>
        <Link href={`/tags/${t}`} text={t} />
      </Tag>
    ))}
  </Tags>
)

PostTags.propTypes = {
  tags: arrayOf(string).isRequired
}

export default PostTags
