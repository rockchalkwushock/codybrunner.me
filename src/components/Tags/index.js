import React from 'react'

import FlexContainer from '../FlexContainer'
import Grid from '../Grid'
import Link from '../Link'

// TODO: Need to create ternary in the event no tags exist which would be like wtf?
// TODO: Styling

const Tags = ({ tags }) => (
  <FlexContainer>
    <Grid>
      <h1>Blog Tags</h1>
      {tags.map(tag => <Link href={`/tags/${tag}`} key={tag} text={tag} />)}
    </Grid>
  </FlexContainer>
)

export default Tags
