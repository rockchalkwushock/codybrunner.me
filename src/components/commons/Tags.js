import React from 'react'
import { arrayOf, string } from 'prop-types'

import List from './List'
import Tag from './Tag'

const Tags = ({ tags }) => (
  <List gap={8} tags>
    {tags.map(t => (
      <Tag key={t} to={`/tags/${t.toLowerCase()}`}>
        {t.toLowerCase()}
      </Tag>
    ))}
  </List>
)

Tags.propTypes = {
  tags: arrayOf(string).isRequired
}

export default Tags
