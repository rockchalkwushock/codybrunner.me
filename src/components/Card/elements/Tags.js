import React from 'react'
import styled from 'styled-components'
import { arrayOf, string } from 'prop-types'

import { Link } from '../../commons'

const Tag = styled.button`
  background-color: ${({ theme }) => theme.site.fg};
  border: none;
  border-radius: 10px;
  box-shadow: 1px 1px 5px 1px rgba(0, 0, 0, 0.7);
  font-size: 0.8rem;
  letter-spacing: 1px;
  margin: 0.25em 0.5em;
  outline: none;
  padding: 0.25em 0.5em;
  transition: all 0.3s ease-in-out;
  :hover {
    transform: scale(1.2);
  }
`

const CardTags = ({ tags }) =>
  tags.map(t => (
    <Tag key={t}>
      <Link href={`/tags/${t}`} text={t} />
    </Tag>
  ))

CardTags.propTypes = {
  tags: arrayOf(string).isRequired
}
export { Tag }
export default CardTags
