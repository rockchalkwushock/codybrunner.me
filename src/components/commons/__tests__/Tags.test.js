import React from 'react'
import { shallow } from 'enzyme'

import Tags from '../Tags'

const tags = ['hello', 'ALL CAPS', 'world']

describe('Component: <Tags />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Tags tags={tags} />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
