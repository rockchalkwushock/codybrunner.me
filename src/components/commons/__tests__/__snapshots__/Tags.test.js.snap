// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Tags /> should render correctly 1`] = `
<List
  flow="row"
  gap={8}
  tags={true}
>
  <Tag
    aria-label="posts for hello"
    key="hello"
    to="/tags/hello"
  >
    hello
  </Tag>
  <Tag
    aria-label="posts for ALL CAPS"
    key="ALL CAPS"
    to="/tags/all caps"
  >
    all caps
  </Tag>
  <Tag
    aria-label="posts for world"
    key="world"
    to="/tags/world"
  >
    world
  </Tag>
</List>
`;
